src/schema/items/T1w.yaml

```diff
@@ -0,0 +1,17 @@
+---
+name: T1-weighted
+class: suffix
+description: |
+  T1-weighted MRI scan.
+# Suffix-specific fields
+extensions:
+  - .nii.gz
+  - .nii
+  - .json
+entities:
```

Comment:
I think this is the main thing that @yarikoptic will object to with this structure. If we have a single file for each suffix, for example, we will have a very large number of files to update any time we change the specification's supported entities.

Reply:
This is especially true for anatomical MRI scans, in which many suffixes have the same rules.

Reply:
On the other hand, I think there's a lot of benefit in having suffix-specific definitions at least. Perhaps we could start with the unfortunate duplication, and then try an inheritance-like procedure as discussed in #588. For example, there could be "parent" YAML files with entities and extensions specified that the "child" YAMLs (like `T1w.yaml`) could inherit from. These parent files would correspond to the subgroups we currently have in our datatype YAML files. Does that make sense?

Reply:
But why to bring in `entities` and `extensions` in here?    
- It is not per se "item"'s property, in particular whenever it might be both a `suffix` and a `datatype` (ref: https://github.com/bids-standard/bids-specification/issues/603#issuecomment-693551460)
- I think records in `items/` should have no "class-specific" attributes and be of a uniform schema (which attributes they could have etc).  I think if those are introduced, establishing later a schema (and a validator) for our schema would be trickier
- What if we keep them ("extensions", "entities") where/as they are  (i.e. `datatypes/anat.yaml` for T1w).  Then files in `items/`  could just provide a file for **every** "name" (well -- "altname") mentioned in the `datatypes/*yaml` file in any section, thus just centralizing their description etc:
  -  Even extensions could get their class (i.e. "extension") and we could have a bunch of (hidden ;))  files (like `.nii.gz.yaml`) which would provide description etc

Reply:
was your comment a reply to mine?  github seems placed it above mine -- just want to make sure

Reply:
https://github.com/bids-standard/bids-specification/pull/609#discussion_r489646301 was before https://github.com/bids-standard/bids-specification/pull/609#discussion_r489667275.

Reply:
I am inclined to think that the entities and extensions _are_ properties of the items. It's just that we need to treat instances where one term refers to two difference classes (e.g., `meg` as a `datatype` and `suffix`) as separate items. There should be a `datatype-meg` and a `suffix-meg`, for example. While that is my preference, though, I'm willing to defer to folks with more experience in this area.

> I think records in items/ should have no "class-specific" attributes and be of a uniform schema (which attributes they could have etc). I think if those are introduced, establishing later a schema (and a validator) for our schema would be trickier

I don't think that's entirely possible. For example, entities are going to have features that are distinct from other classes, such as whether they take an `index` or a `label`, as well as their "entity-key". Metadata fields, at minimum, need "units". I think those are features of the actual items, rather than how they interact with other classes.

> What if we keep them ("extensions", "entities") where/as they are (i.e. datatypes/anat.yaml for T1w). Then files in items/ could just provide a file for every "name" (well -- "altname") mentioned in the datatypes/*yaml file in any section, thus just centralizing their description etc:

Well, on the bright side we can always change things later, so I'm willing to give it a try. I can start with the filename, the description, and a "display-name" (per @effigies suggestion) for each of the suffixes. If we want to ultimately add the entities, extensions, and possibly metadata fields later on, that should be easy enough.

Reply:
eh the call ended while I was staring at this PR and thought to may be chat about it further (sorry for being slow with replies on github): I think if we make this `items/` folder into `terms/` it could be pretty much what [BIDS_Terms folder](https://github.com/nqueder/bids_terms_to_markdown_table/tree/master/BIDS_Terms) is (just nicer yamls, from which jsonlds would be generated) with just with descriptions tuned up to not provide any "BIDS formatting" specifics, and rather describing the actual generic concept (subject, session, MR enhancement contrast agent, etc). Then it would be for schema "upstairs" (in `entities/` etc) to use the terms provided under `terms/`, so any "word" used in the schema has clear mapping to a term.
It would be ok for the same term to be used in different contexts (`datatype/` folder vs a filename `_suffix`).  If there is a need to provide a "concept" of a composite term + context where it is used, I think we (or whoever needs them) could automatically "mint" them, at the level of composition they need. So there could be a concept of `sub-*/[ses-*/]<datatype-term>/` corresponding to a "BIDS folder with <datatype-term> data for a single acquisition session of a single subject", thus reflecting layout limitation of BIDS (i.e. we cannot have a folder with eeg data for all subjects). 
I think it also boils down to "contexts" to provide mapping if there is no 1-to-1 mapping. E.g. (hypothetical since no concrete case comes to mind ATM; and may be not good here since IMHO the example term is a composition of modality + concept): If there was an `_nchan-<number>` entity to signal number of channels in the file, then it would correspond to either ECGChannelCount or EEGChannelCount etc depending on the context -- datatype for which it is listed. So we would then need to be able somehow to say that entity points to different terms based on some other level (datatype ATM).  For that we could provision that any entity listing does not only list its "use" (optional vs required; may be there is a better name) but also a "term". E.g. `datatypes/eeg.yaml` could have
```yaml
  ...
  entities:
    sub: required
    ses: optional
    task: required
    acq: optional
    run: optional
    nchan:
      use: optional
      term: EEGChannelCount
```
and allow similar gimmick for any place where term could be used and we need to remap depending on the context.

To me, it is not yet totally clear on where we would need such "composite" terms. Since even in the case of EEGChannelCount, [it is sameAs](https://github.com/nqueder/bids_terms_to_markdown_table/blob/master/BIDS_Terms/ECGChannelCount.jsonld) a generic "http://purl.org/nidash/nidm#NumberOfChannels", so I think overall it might be a utopia to fight combinatorics at the level of terms ;), and we should get away from it.